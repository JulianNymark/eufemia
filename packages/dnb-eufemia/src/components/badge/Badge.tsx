import React from 'react'
import classnames from 'classnames'

// Components
import { createSpacingClasses } from '../space/SpacingHelper'
import { createSkeletonClass } from '../skeleton/SkeletonHelper'

// Shared
import Context from '../../shared/Context'
import { ISpacingProps, SkeletonTypes } from '../../shared/interfaces'
import { extendPropsWithContext } from '../../shared/component-helper'

export interface BadgeProps {
  /**
   * Custom className on the component root
   * Default: null
   */
  className?: string

  /**
   * Skeleton should be applied when loading content
   * Default: null
   */
  skeleton?: SkeletonTypes

  /**
   * The content to display the badge on top of.
   * Default: null
   */
  children?: React.ReactNode

  /**
   * The content of the component.
   * Default: null
   */
  content?: string | number | React.ReactNode

  /**
   * The size of the component.
   * Default: medium.
   */
  size?: 'small' | 'medium' | 'large' | 'x-large'

  /**
   * The variant of the component.
   * Default: primary.
   */
  variant?: 'primary' | 'secondary' | 'tertiary'

  /**
   * The vertical positioning of the component.
   * Default: bottom.
   */
  vertical?: 'bottom' | 'top'

  /**
   * The horizontal positioning of the component.
   * Default: right.
   */
  horizontal?: 'left' | 'right'
}

export const defaultProps = {
  className: null,
  variant: 'primary',
  skeleton: false,
  children: null,
  content: null,
  vertical: 'bottom',
  horizontal: 'right',
  size: 'medium',
}

function Badge(localProps: BadgeProps & ISpacingProps) {
  // Every component should have a context
  const context = React.useContext(Context)
  // Extract additional props from global context
  const {
    className,
    children,
    skeleton,
    variant,
    horizontal,
    vertical,
    content: contentProp,
    size,
    ...props
  } = extendPropsWithContext(
    { ...defaultProps, ...localProps },
    defaultProps,
    context?.Badge
  )

  const skeletonClasses = createSkeletonClass('shape', skeleton, context)
  const spacingClasses = createSpacingClasses(props)
  const contentIsNum = typeof contentProp === 'number'
  const hasText =
    contentProp && (contentIsNum || typeof contentProp === 'string')

  const content = contentIsNum && contentProp > 99 ? '99+' : contentProp

  return (
    <span className="dnb-badge__root" data-testid="badge-root">
      {children}
      <span
        className={classnames(
          'dnb-badge',
          `dnb-badge--${variant || 'primary'}`,
          `dnb-badge--horizontal-${horizontal || 'right'}`,
          `dnb-badge--vertical-${vertical || 'bottom'}`,
          `dnb-badge--size-${size || 'medium'}`,
          hasText && 'dnb-badge--has-text',
          skeletonClasses,
          spacingClasses,
          className
        )}
        data-testid="badge"
        {...props}
      >
        {content}
      </span>
    </span>
  )
}

export default Badge
