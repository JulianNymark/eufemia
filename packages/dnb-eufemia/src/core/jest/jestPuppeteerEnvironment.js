/**
 * Jest Setup for Screenshot testing
 *
 */

const NodeEnvironment = require('jest-environment-node')
const fs = require('fs-extra')
const path = require('path')
const chalk = require('chalk')
const puppeteer = require('puppeteer')
const { DIR } = require('./jestSetupScreenshots').config

class PuppeteerEnvironment extends NodeEnvironment {
  constructor(config) {
    super(config)
  }

  async setup() {
    await super.setup()

    // get the wsEndpoint
    const wsEndpoint = await fs.readFile(
      path.join(DIR, 'wsEndpoint'),
      'utf8'
    )
    if (!wsEndpoint) {
      throw new Error('wsEndpoint not found')
    }

    // connect to puppeteer
    this.global.__BROWSER__ = await puppeteer.connect({
      browserWSEndpoint: wsEndpoint,
    })

    this.global.__PAGE__ = await this.global.__BROWSER__.newPage()
  }

  async handleTestEvent(event, state) {
    if (event.name === 'test_fn_failure') {
      this.global.__EVENT_FAILURE__ = true

      const { currentlyRunningTest } = state
      console.log(
        chalk.yellow(
          `Retry attempt #${currentlyRunningTest.invocations}: ${currentlyRunningTest.parent.name} / ${currentlyRunningTest.name}`
        )
      )
    }
  }

  async teardown() {
    await this.global.__PAGE__.close()
    this.global.__PAGE__ = null

    await super.teardown()
  }

  runScript(script) {
    return super.runScript(script)
  }
}

module.exports = PuppeteerEnvironment
