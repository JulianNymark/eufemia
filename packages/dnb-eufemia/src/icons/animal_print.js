/** This file is auto generated by convertSvgToJsx.js */

import React from 'react'

function animal_print(props) {
  return (
    <svg
      width={16}
      height={16}
      viewBox="0 0 16 16"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M10.3333 9.82791C10.3333 8.51128 9.36757 7.21456 8 7.21456C6.63243 7.21456 5.66667 8.51128 5.66667 9.82791C4.57261 9.82791 3.8 10.8653 3.8 11.9186C3.8 13.5459 5.44629 14.5352 6.84267 13.6932C7.55741 13.277 8.44197 13.277 9.15671 13.6932C10.5107 14.509 12.2 13.6024 12.2 11.9186C12.2 10.8653 11.4274 9.82791 10.3333 9.82791Z"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M2.4 9.26877C3.1732 9.26877 3.8 8.49056 3.8 7.53059C3.8 6.57062 3.1732 5.7924 2.4 5.7924C1.6268 5.7924 1 6.57062 1 7.53059C1 8.49056 1.6268 9.26877 2.4 9.26877Z"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M5.66667 5.47637C6.43987 5.47637 7.06667 4.69816 7.06667 3.73819C7.06667 2.77821 6.43987 2 5.66667 2C4.89347 2 4.26667 2.77821 4.26667 3.73819C4.26667 4.69816 4.89347 5.47637 5.66667 5.47637Z"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M10.3333 5.47637C11.1065 5.47637 11.7333 4.69816 11.7333 3.73819C11.7333 2.77821 11.1065 2 10.3333 2C9.56013 2 8.93333 2.77821 8.93333 3.73819C8.93333 4.69816 9.56013 5.47637 10.3333 5.47637Z"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M13.6 9.26877C14.3732 9.26877 15 8.49056 15 7.53059C15 6.57062 14.3732 5.7924 13.6 5.7924C12.8268 5.7924 12.2 6.57062 12.2 7.53059C12.2 8.49056 12.8268 9.26877 13.6 9.26877Z"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  )
}

export default animal_print
