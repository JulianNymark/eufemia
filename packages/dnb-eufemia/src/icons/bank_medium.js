/** This file is auto generated by convertSvgToJsx.js */

import React from 'react'

function bank_medium(props) {
  return (
    <svg
      width={24}
      height={24}
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M1.5 22.5001H22.5M21 10.5001V19.5001H18V10.5001M13.5 10.5001V19.5001H10.5V10.5001M6 10.5001V19.5001H3V10.5001M22.5 7.50014H1.5L11.189 1.73814C11.4309 1.58266 11.7124 1.5 12 1.5C12.2876 1.5 12.5691 1.58266 12.811 1.73814L22.5 7.50014Z"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  )
}

export default bank_medium
