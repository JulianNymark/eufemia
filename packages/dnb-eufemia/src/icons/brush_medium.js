/** This file is auto generated by convertSvgToJsx.js */

import React from 'react'

function brush_medium(props) {
  return (
    <svg
      width={24}
      height={24}
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M7.67575 14.2849C7.23258 14.0945 6.75278 13.9896 6.26159 13.981C5.29008 13.9638 4.34985 14.3244 3.63898 14.9868C3.28416 15.3486 3.00509 15.7776 2.8181 16.2486C2.63111 16.7196 2.53995 17.2232 2.54998 17.7298C2.39998 21.4848 0.987976 22.9418 0.987976 22.9418C2.42948 23.192 3.91054 23.0803 5.29827 22.6169C6.686 22.1536 7.93698 21.3529 8.93898 20.2868C9.60138 19.576 9.96199 18.6357 9.94485 17.6642C9.9369 17.2137 9.84802 16.7728 9.68646 16.361M7.67575 14.2849C8.10909 14.4711 8.5074 14.739 8.84708 15.0787C9.21555 15.4472 9.49961 15.8846 9.68646 16.361M7.67575 14.2849L8.49997 13.2218L10.8141 10.4998M9.68646 16.361L10.817 15.3997L13.4568 13.1415M10.8141 10.4998L18.133 1.89083C18.381 1.59996 18.6866 1.36357 19.0305 1.19652C19.3743 1.02947 19.7491 0.93538 20.131 0.920165C20.513 0.904949 20.894 0.968938 21.2501 1.10811C21.6061 1.24728 21.9296 1.45862 22.2 1.72883C22.4709 1.99924 22.6829 2.32285 22.8225 2.67925C22.9621 3.03565 23.0264 3.41712 23.0113 3.7996C22.9962 4.18208 22.902 4.55728 22.7346 4.90155C22.5673 5.24581 22.3304 5.55166 22.039 5.79983L13.4568 13.1415M10.8141 10.4998L13.4568 13.1415"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  )
}

export default brush_medium
