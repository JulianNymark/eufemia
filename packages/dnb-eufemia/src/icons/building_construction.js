/** This file is auto generated by convertSvgToJsx.js */

import React from 'react'

function building_construction(props) {
  return (
    <svg
      width={16}
      height={16}
      viewBox="0 0 16 16"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M2.65404 8.95625V13.2159H5.96213V9.79308C5.96213 9.34271 6.3324 8.97761 6.78915 8.97761H7.78765M1 7.14078L2.65441 5.50965M12.044 5.79845L7.78765 1.48876C7.46469 1.17041 6.94119 1.17041 6.61824 1.48876L4.72196 3.35853M11.7328 14.75V10.9262M2.65441 5.50965V1.75H4.72196V3.35853M2.65441 5.50965L4.72196 3.35853M11.7328 10.9262H13.133V11.0797C13.133 11.2424 13.2791 11.3865 13.4442 11.3865H15V8.16488H13.4442C13.2791 8.16488 13.133 8.30895 13.133 8.4717V8.62511H11.266C10.7708 8.62511 10.296 8.81906 9.94584 9.1643C9.59571 9.50954 9.399 9.97778 9.399 10.466V11.465C9.40017 11.6832 9.69418 11.7698 9.82095 11.6012C10.0256 11.2476 10.3922 10.9947 10.7993 10.9262H11.7328Z"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  )
}

export default building_construction
