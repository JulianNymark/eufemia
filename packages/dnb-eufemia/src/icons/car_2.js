/** This file is auto generated by convertSvgToJsx.js */

import React from 'react'

function car_2(props) {
  return (
    <svg
      width={16}
      height={16}
      viewBox="0 0 16 16"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M12.1801 7.60328C12.9248 7.60228 13.6395 7.95614 14.168 8.58748C14.6964 9.21881 14.9956 10.0763 15 10.9723C15 11.2701 14.9017 11.5557 14.7266 11.7663C14.5516 11.9769 14.3142 12.0953 14.0667 12.0953C13.7414 12.0953 13.4162 12.0953 13.0909 12.0953M12.1801 7.60328C12.1801 7.60328 10.7802 2.5498 7.04689 2.5498C6.9406 2.5498 6.834 2.5527 6.72727 2.55862M12.1801 7.60328L6.72727 7.59982M2.72964 12.0953C2.72964 12.7982 3.29946 13.368 4.00237 13.368C4.6514 13.368 5.18696 12.8822 5.26525 12.2543M2.72964 12.0953C2.72964 11.3924 3.29946 10.8225 4.00237 10.8225C4.70527 10.8225 5.27509 11.3924 5.27509 12.0953C5.27509 12.1491 5.27174 12.2022 5.26525 12.2543M2.72964 12.0953L1.91365 12.0953C1.79027 12.0951 1.66819 12.0649 1.55476 12.0065C1.44133 11.9481 1.3389 11.8626 1.25362 11.7554C1.16835 11.6481 1.10199 11.5211 1.05856 11.3822C1.01512 11.2432 0.995496 11.0951 1.00087 10.9468C1.04887 9.65985 1.20924 8.55065 1.45765 7.59982M13.0909 12.0953C13.0909 12.7982 12.5211 13.368 11.8182 13.368C11.1692 13.368 10.6336 12.8822 10.5553 12.2543M13.0909 12.0953C13.0909 11.3924 12.5211 10.8225 11.8182 10.8225C11.1153 10.8225 10.5455 11.3924 10.5455 12.0953C10.5455 12.1491 10.5488 12.2022 10.5553 12.2543M6.72727 2.55862V7.59982M6.72727 2.55862C4.58888 2.67718 2.39564 4.00948 1.45765 7.59982M6.72727 7.59982H1.45765M5.26525 12.2543H10.5553"
        stroke="black"
        strokeWidth={1.5}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  )
}

export default car_2
