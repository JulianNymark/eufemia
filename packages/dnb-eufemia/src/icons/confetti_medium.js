/** This file is auto generated by convertSvgToJsx.js */

import React from 'react'

function confetti_medium(props) {
  return (
    <svg
      width={25}
      height={25}
      viewBox="0 0 25 25"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M0.901888 22.5406L1.57713 22.9022L1.57759 22.9014L0.901888 22.5406ZM1.10308 23.8969L0.560695 24.4377L0.562238 24.4393L1.10308 23.8969ZM2.46035 24.0981L2.09995 23.4222L2.09942 23.4225L2.46035 24.0981ZM10.0628 14.9424L9.52095 15.4837L9.52121 15.484L10.0628 14.9424ZM7.97227 9.26714L7.43094 8.72525L7.42982 8.72637L7.97227 9.26714ZM13.6465 11.3577L14.1882 10.8162L14.1878 10.8158L13.6465 11.3577ZM16.5112 12.1686C16.2119 12.4675 16.2117 12.9525 16.5106 13.2518C16.8096 13.5511 17.2946 13.5513 17.5938 13.2524L16.5112 12.1686ZM23.6785 13.2524C23.9778 13.5513 24.4627 13.5511 24.7617 13.2518C25.0607 12.9525 25.0604 12.4675 24.7611 12.1686L23.6785 13.2524ZM11.7328 7.3915C11.4336 7.69055 11.4335 8.17553 11.7325 8.47473C12.0316 8.77393 12.5166 8.77406 12.8158 8.47501L11.7328 7.3915ZM12.8162 0.224623C12.5172 -0.0746559 12.0322 -0.074906 11.733 0.224064C11.4337 0.523034 11.4334 1.00801 11.7324 1.30729L12.8162 0.224623ZM18.4731 7.93253C18.0718 8.06613 17.8547 8.49981 17.9883 8.90118C18.1219 9.30256 18.5556 9.51963 18.957 9.38603L18.4731 7.93253ZM22.5406 8.19318C22.942 8.05958 23.1591 7.6259 23.0255 7.22453C22.8918 6.82315 22.4582 6.60608 22.0568 6.73968L22.5406 8.19318ZM15.5997 6.02943C15.466 6.43074 15.6829 6.86452 16.0842 6.99829C16.4855 7.13206 16.9193 6.91518 17.053 6.51386L15.5997 6.02943ZM18.2479 2.92919C18.3817 2.52787 18.1648 2.09409 17.7635 1.96032C17.3622 1.82655 16.9284 2.04344 16.7946 2.44476L18.2479 2.92919ZM15.7919 8.29826L15.2502 7.75665L15.2489 7.75803L15.7919 8.29826ZM16.5139 8.29826L17.0569 7.75803L17.0555 7.75665L16.5139 8.29826ZM8.92318 6.0343L8.38156 5.49269L8.38018 5.49407L8.92318 6.0343ZM8.20114 6.0343L8.74413 5.49407L8.74275 5.49269L8.20114 6.0343ZM20.9481 4.77762L21.4883 5.32063L21.4897 5.31923L20.9481 4.77762ZM20.226 4.77762L19.6844 5.31924L19.6858 5.32062L20.226 4.77762ZM6.93698 9.60906L0.226186 22.1799L1.57759 22.9014L8.28838 10.3305L6.93698 9.60906ZM0.226645 22.1791C-0.172173 22.9239 -0.0348617 23.8405 0.560697 24.4377L1.64546 23.3561C1.52409 23.2343 1.49799 23.05 1.57713 22.9022L0.226645 22.1791ZM0.562238 24.4393C1.1598 25.0351 2.07617 25.1717 2.82128 24.7737L2.09942 23.4225C1.9499 23.5024 1.76533 23.4756 1.64392 23.3545L0.562238 24.4393ZM2.82076 24.7739L15.4304 18.0499L14.7096 16.6981L2.09995 23.4222L2.82076 24.7739ZM15.1954 16.4913C15.0391 16.6476 14.566 16.8009 13.5978 16.4443C12.6969 16.1123 11.6166 15.413 10.6044 14.4007L9.52121 15.484C10.6537 16.6164 11.9157 17.4571 13.0683 17.8817C14.1537 18.2816 15.4453 18.4079 16.2786 17.5745L15.1954 16.4913ZM10.6047 14.401C9.5928 13.3882 8.89361 12.3081 8.56168 11.4072C8.20514 10.4395 8.35777 9.96535 8.51472 9.80791L7.42982 8.72637C6.59818 9.56058 6.72425 10.8512 7.12423 11.9368C7.5488 13.0891 8.38918 14.3509 9.52095 15.4837L10.6047 14.401ZM8.5136 9.80903C8.66956 9.65324 9.14283 9.50015 10.1117 9.85728C11.0131 10.1895 12.0934 10.8888 13.1051 11.8996L14.1878 10.8158C13.0559 9.68497 11.7939 8.84468 10.6415 8.4199C9.5566 8.02001 8.2646 7.89245 7.43094 8.72525L8.5136 9.80903ZM13.1047 11.8992C14.117 12.9119 14.8164 13.9924 15.1484 14.8936C15.505 15.8619 15.3517 16.335 15.1954 16.4913L16.2786 17.5745C17.112 16.7412 16.9857 15.4495 16.5858 14.3641C16.1613 13.2114 15.3206 11.9491 14.1882 10.8162L13.1047 11.8992ZM17.5938 13.2524C19.274 11.5739 21.9983 11.5739 23.6785 13.2524L24.7611 12.1686C22.4829 9.89264 18.7895 9.89264 16.5112 12.1686L17.5938 13.2524ZM12.8158 8.47501C15.0945 6.19741 15.0939 2.50472 12.8162 0.224623L11.7324 1.30729C13.4131 2.98973 13.4125 5.71268 11.7328 7.3915L12.8158 8.47501ZM18.957 9.38603L22.5406 8.19318L22.0568 6.73968L18.4731 7.93253L18.957 9.38603ZM17.053 6.51386L18.2479 2.92919L16.7946 2.44476L15.5997 6.02943L17.053 6.51386ZM15.9723 8.48073C16.0723 8.38075 16.2335 8.38075 16.3335 8.48073L15.2502 9.56395C15.7485 10.0622 16.5572 10.0622 17.0555 9.56395L15.9723 8.48073ZM16.3335 8.48073C16.4342 8.58147 16.4322 8.74062 16.3349 8.83848L15.2489 7.75803C14.7532 8.25624 14.7512 9.06492 15.2502 9.56395L16.3335 8.48073ZM16.3335 8.83987C16.2335 8.93985 16.0723 8.93985 15.9723 8.83987L17.0555 7.75665C16.5572 7.25837 15.7485 7.25837 15.2502 7.75665L16.3335 8.83987ZM15.9709 8.83848C15.8735 8.74062 15.8715 8.58147 15.9723 8.48073L17.0555 9.56395C17.5545 9.06492 17.5526 8.25624 17.0569 7.75803L15.9709 8.83848ZM8.38018 5.49407C8.48047 5.39327 8.64384 5.39327 8.74413 5.49407L7.65814 6.57452C8.15673 7.07567 8.96759 7.07567 9.46617 6.57452L8.38018 5.49407ZM8.74275 5.49269C8.84273 5.59266 8.84273 5.7539 8.74275 5.85387L7.65952 4.77065C7.16125 5.26892 7.16125 6.07764 7.65952 6.57591L8.74275 5.49269ZM8.74275 5.85387C8.64277 5.95385 8.48154 5.95385 8.38157 5.85387L9.46479 4.77065C8.96652 4.27237 8.1578 4.27237 7.65952 4.77065L8.74275 5.85387ZM8.38157 5.85387C8.28159 5.7539 8.28159 5.59266 8.38157 5.49269L9.46479 6.57591C9.96306 6.07764 9.96306 5.26892 9.46479 4.77065L8.38157 5.85387ZM20.4078 4.23462C20.5075 4.13548 20.6666 4.13548 20.7662 4.23462L19.6858 5.32062C20.1838 5.81602 20.9903 5.81602 21.4883 5.32062L20.4078 4.23462ZM20.7676 4.23601C20.8676 4.33599 20.8676 4.49722 20.7676 4.59719L19.6844 3.51397C19.1861 4.01224 19.1861 4.82096 19.6844 5.31923L20.7676 4.23601ZM20.7676 4.59719C20.6677 4.69717 20.5064 4.69717 20.4065 4.59719L21.4897 3.51397C20.9914 3.0157 20.1827 3.0157 19.6844 3.51397L20.7676 4.59719ZM20.4065 4.59719C20.3065 4.49722 20.3065 4.33599 20.4065 4.23601L21.4897 5.31923C21.9879 4.82096 21.9879 4.01224 21.4897 3.51397L20.4065 4.59719ZM19.7292 14.4527C19.8292 14.3528 19.9905 14.3528 20.0904 14.4527L19.0072 15.536C19.5055 16.0342 20.3142 16.0342 20.8125 15.536L19.7292 14.4527ZM20.0904 14.4527C20.1904 14.5527 20.1904 14.7139 20.0904 14.8139L19.0072 13.7307C18.5089 14.229 18.5089 15.0377 19.0072 15.536L20.0904 14.4527ZM20.0904 14.8139C19.9905 14.9139 19.8292 14.9139 19.7292 14.8139L20.8125 13.7307C20.3142 13.2324 19.5055 13.2324 19.0072 13.7307L20.0904 14.8139ZM19.7292 14.8139C19.6293 14.7139 19.6293 14.5527 19.7292 14.4527L20.8125 15.536C21.3107 15.0377 21.3107 14.229 20.8125 13.7307L19.7292 14.8139Z"
        fill="black"
      />
    </svg>
  )
}

export default confetti_medium
