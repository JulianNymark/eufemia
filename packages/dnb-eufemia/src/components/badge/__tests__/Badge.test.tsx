import React from 'react'
import { render, screen, within } from '@testing-library/react'
import Badge from '../Badge'
import { confetti as Confetti } from '../../../icons'
import Icon from '../../Icon'
import Avatar from '../../Avatar'

import { loadScss, axeComponent } from '../../../core/jest/jestSetup'

describe('Badge', () => {
  it('renders without properties', () => {
    render(<Badge />)

    expect(screen.queryByTestId('badge')).not.toBeNull()
  })

  it('renders content as text', () => {
    const string = 'A'
    render(<Badge content={string} />)

    expect(screen.queryByTestId('badge').textContent).toBe(string)
  })

  it('renders content as number', () => {
    const number = 1
    render(<Badge content={number} />)

    expect(screen.queryByTestId('badge').textContent).toBe(
      number.toString()
    )
  })

  it('renders class "has-text" when string or number as content', () => {
    const string = 'BC'

    render(<Badge content={string} />)

    expect(
      screen
        .queryByTestId('badge')
        .classList.contains('dnb-badge--has-text')
    ).toBe(true)
  })

  it('renders 99+ when content is over 99 number', () => {
    const number = 100
    render(<Badge content={number} />)

    expect(screen.queryByTestId('badge').textContent).toBe('99+')
  })

  it('renders children/Icon as content', () => {
    render(
      <Badge
        size="large"
        content={<Icon icon={Confetti} data-testid="confetti-icon" />}
        variant="secondary"
      >
        <Avatar>A</Avatar>
      </Badge>
    )

    const badgeRoot = screen.queryByTestId('badge-root')
    expect(within(badgeRoot).queryByTestId('confetti-icon')).not.toBeNull()
  })
  describe('default values', () => {
    it('has variant primary as default', () => {
      render(<Badge />)
      expect(
        screen
          .queryByTestId('badge')
          .classList.contains('dnb-badge--primary')
      ).toBe(true)
    })

    it('has horizontal right as default', () => {
      render(<Badge />)
      expect(
        screen
          .queryByTestId('badge')
          .classList.contains('dnb-badge--horizontal-right')
      ).toBe(true)
    })

    it('has vertical bottom as default', () => {
      render(<Badge />)
      expect(
        screen
          .queryByTestId('badge')
          .classList.contains('dnb-badge--vertical-bottom')
      ).toBe(true)
    })

    it('has size medium as default', () => {
      render(<Badge />)
      expect(
        screen
          .queryByTestId('badge')
          .classList.contains('dnb-badge--size-medium')
      ).toBe(true)
    })
  })
})

describe('Badge aria', () => {
  it('should validate', async () => {
    const Component = render(<Badge content="1" />)
    expect(await axeComponent(Component)).toHaveNoViolations()
  })
})

describe('Badge scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-badge.scss'))
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-badge-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
