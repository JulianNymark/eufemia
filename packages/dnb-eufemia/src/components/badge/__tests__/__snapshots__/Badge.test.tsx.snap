// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge scss have to match default theme snapshot 1`] = `
"/*
* Badge theme
*
*/
/**
 * This file is only used to make themes independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
.dnb-badge {
  color: var(--color-pistachio); }
  .dnb-badge--primary {
    background-color: var(--color-sea-green); }
  .dnb-badge--secondary {
    background-color: var(--color-emerald-green); }
  .dnb-badge--tertiary {
    background-color: var(--color-mint-green);
    color: var(--color-emerald-green); }
"
`;

exports[`Badge scss have to match snapshot 1`] = `
"/*
* DNB Badge
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
.dnb-badge {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-basis);
  font-size: var(--font-size-small);
  font-style: normal;
  line-height: var(--line-height-basis);
  color: var(--color-black-80, #333);
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  /**
  * Ensure consistency and use the same as HTML reset -> html {...}
  * between base and code package
  */
  -moz-tab-size: 4;
  tab-size: 4;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  word-break: break-word;
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-badge *,
  .dnb-badge ::before,
  .dnb-badge ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-badge ::before,
  .dnb-badge ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
* Badge component
*
*/
:root {
  --badge-font-size--small: 0.5rem;
  --badge-font-size--medium: 0.75rem;
  --badge-font-size--large: var(--font-size-x-small);
  --badge-font-size--x-large: var(--font-size-basis);
  --badge-line-height--small: var(--line-height-xx-small);
  --badge-line-height--medium: var(--line-height-xx-small);
  --badge-line-height--large: var(--line-height-x-small);
  --badge-line-height--x-large: var(--line-height-basis);
  --badge-width--small: 0.5rem;
  --badge-height--small: 0.5rem;
  --badge-border-radius--small: 0.25rem;
  --badge-width--medium: 0.75rem;
  --badge-height--medium: 0.75rem;
  --badge-border-radius--medium: 0.375rem;
  --badge-width--large: 1.5rem;
  --badge-height--large: 1.5rem;
  --badge-border-radius--large: 0.75rem;
  --badge-width--x-large: 2rem;
  --badge-height--x-large: 2rem;
  --badge-border-radius--x-large: 1rem; }

.dnb-badge {
  display: flex;
  flex-direction: row;
  white-space: nowrap;
  flex-wrap: wrap;
  justify-content: center;
  align-content: center;
  align-items: center;
  position: absolute;
  box-sizing: border-box;
  overflow: hidden;
  z-index: 2; }
  .dnb-badge__root {
    position: relative;
    display: inline-flex;
    vertical-align: middle;
    flex-shrink: 0; }
  .dnb-badge,
  .dnb-core-style .dnb-badge {
    line-height: var(--button-height); }
  .dnb-badge--size-small {
    width: var(--badge-width--small);
    height: var(--badge-height--small);
    border-radius: var(--badge-border-radius--small);
    font-size: var(--badge-font-size--small); }
    .dnb-badge--size-small,
    .dnb-core-style .dnb-badge--size-small {
      line-height: var(--badge-line-height--small); }
  .dnb-badge--size-medium {
    width: var(--badge-width--medium);
    height: var(--badge-height--medium);
    border-radius: var(--badge-border-radius--medium);
    font-size: var(--badge-font-size--medium); }
    .dnb-badge--size-medium,
    .dnb-core-style .dnb-badge--size-medium {
      line-height: var(--badge-line-height--medium); }
  .dnb-badge--size-large {
    width: var(--badge-width--large);
    height: var(--badge-height--large);
    border-radius: var(--badge-border-radius--large);
    font-size: var(--badge-font-size--large); }
    .dnb-badge--size-large,
    .dnb-core-style .dnb-badge--size-large {
      line-height: var(--badge-line-height--large); }
  .dnb-badge--size-x-large {
    width: var(--badge-width--x-large);
    height: var(--badge-height--x-large);
    border-radius: var(--badge-border-radius--x-large);
    font-size: var(--badge-font-size--x-large); }
    .dnb-badge--size-x-large,
    .dnb-core-style .dnb-badge--size-x-large {
      line-height: var(--badge-line-height--x-large); }
  .dnb-badge--has-text.dnb-badge--size-small {
    min-width: var(--badge-width--small);
    width: auto;
    padding: 0 1px; }
  .dnb-badge--has-text.dnb-badge--size-medium {
    min-width: var(--badge-width--medium);
    width: auto;
    padding: 0 2px; }
  .dnb-badge--has-text.dnb-badge--size-large {
    min-width: var(--badge-width--large);
    width: auto;
    padding: 0 6px; }
  .dnb-badge--has-text.dnb-badge--size-x-large {
    min-width: var(--badge-width--x-large);
    width: auto;
    padding: 0 6px; }
  .dnb-badge--horizontal-left {
    left: 0; }
  .dnb-badge--horizontal-right {
    right: 0; }
  .dnb-badge--vertical-bottom {
    bottom: 0; }
  .dnb-badge--vertical-top {
    top: 0; }
"
`;
